### ISSUE TITLE
MirroredTypeException on @SeeAlso annotations

### ISSUE THREAD

**Issue by aburkholder-vgc** [stoicflame/enunciate#228]
Using maven 3, jdk6
No Enunciate settings file.
POM settings:
```XML
			<plugin>
				<groupId>com.webcohesion.enunciate</groupId>
				<artifactId>enunciate-maven-plugin</artifactId>
				<version>2.1.1</version>
				<configuration>
					<docsDir>${project.build.directory}/target/enunciate/docs</docsDir>
 					<buildDir>${project.build.directory}/target/enunciate/build</buildDir>
					<exportsDir>${project.build.directory}/target/enunciate/exports</exportsDir>
				</configuration>
				<executions>
					<execution>
						<goals>
							<goal>docs</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
```

Stack trace below. If you need the WSDL or generated objects I will have to obfuscate them for you ...
```ObjDump
Dec 1, 2015 1:40:53 PM org.reflections.Reflections scan
INFO: Reflections took 3500 ms to scan 185 urls, producing 1165 keys and 1165 values 
javax.lang.model.type.MirroredTypeException: Attempt to access Class object for TypeMirror com.jaxb.xjc.generated.source.SomeOjbect
	at com.sun.tools.javac.model.AnnotationProxyMaker$MirroredTypeExceptionProxy.generateException(AnnotationProxyMaker.java:282)
	at sun.reflect.annotation.AnnotationInvocationHandler.invoke(AnnotationInvocationHandler.java:57)
	at com.sun.proxy.$Proxy86.value(Unknown Source)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.addSeeAlsoTypeDefinitions(EnunciateJaxbContext.java:761)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:593)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:379)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:840)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:805)
	at com.webcohesion.enunciate.javac.decorations.type.DecoratedDeclaredType.accept(DecoratedDeclaredType.java:83)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.addReferencedTypeDefinitions(EnunciateJaxbContext.java:703)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:639)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:379)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:840)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:805)
	at com.webcohesion.enunciate.javac.decorations.type.DecoratedDeclaredType.accept(DecoratedDeclaredType.java:83)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.addReferencedTypeDefinitions(EnunciateJaxbContext.java:703)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.addReferencedTypeDefinitions(EnunciateJaxbContext.java:692)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:596)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:379)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:840)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:805)
	at com.webcohesion.enunciate.javac.decorations.type.DecoratedDeclaredType.accept(DecoratedDeclaredType.java:83)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.addReferencedTypeDefinitions(EnunciateJaxbContext.java:703)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.addReferencedTypeDefinitions(EnunciateJaxbContext.java:692)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:596)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:379)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:840)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:805)
	at com.webcohesion.enunciate.javac.decorations.type.DecoratedDeclaredType.accept(DecoratedDeclaredType.java:83)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.addReferencedTypeDefinitions(EnunciateJaxbContext.java:703)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.addReferencedTypeDefinitions(EnunciateJaxbContext.java:465)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:445)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:424)
	at com.webcohesion.enunciate.modules.jaxb.JaxbModule.addPotentialJaxbElement(JaxbModule.java:110)
	at com.webcohesion.enunciate.modules.jaxws.JaxwsModule.call(JaxwsModule.java:152)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
	at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
	at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
	at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
	at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
	at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
	at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
	at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918)
	at java.lang.Thread.run(Thread.java:662)
Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class
	at rx.exceptions.OnErrorThrowable.addValueAsLastCause(OnErrorThrowable.java:101)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:81)
	... 15 more
[WARNING] [ENUNCIATE] [javac] 
[WARNING] [ENUNCIATE] [javac] 
[WARNING] [ENUNCIATE] [javac] An annotation processor threw an uncaught exception.
```

**Comment by stoicflame** [comment]
Thanks for the report. Assigned to Enunciate 2.2.

**Comment by stoicflame** [comment]
Okay, this error is really weird.

The stack trace points to [EnunciateJaxbContext.java:761](https://github.com/stoicflame/enunciate/blob/v2.1.1/jaxb/src/main/java/com/webcohesion/enunciate/modules/jaxb/EnunciateJaxbContext.java#L761), but that exception is [clearly getting caught on line 766](https://github.com/stoicflame/enunciate/blob/v2.1.1/jaxb/src/main/java/com/webcohesion/enunciate/modules/jaxb/EnunciateJaxbContext.java#L766) (`MirroredTypeException` extends `MirroredTypesException`). So I have no idea how you're even seeing this error.

I guess I'm going to need that obfuscated WSDL after all...

**Comment by stoicflame** [comment]
No update on this issue. Without a way to reproduce it, I had to take a shot in the dark at e6fabf3, which will be included in 2.3.

I'd be happy to re-open as needed if more information is provided on how to reproduce the issue.

**Comment by stoicflame** [comment]
[Enunciate 2.3 has been released](https://github.com/stoicflame/enunciate/releases/tag/v2.3.0).
